doc/userguide/ug2html.py

```diff
@@ -248,20 +255,20 @@ def copy(source, dest):
     with open(ugpath) as infile:
         content = link_regexp.sub(replace_links, infile.read())
     with open(os.path.join(outdir, os.path.basename(ugpath)), 'wb') as outfile:
-        outfile.write(content)
-    print os.path.abspath(outfile.name)
+        outfile.write(content.encode(sys.getdefaultencoding()))
```

Comment:
When working with various operating systems, we definitely don't want outputs to be written using OS default encoding. Should be using UTF-8 and it should be given as an argument to both of the `open()` calls above. Also, no need to open text files in binary mode.